txscript/pkscript.go

```diff
@@ -0,0 +1,194 @@
+package txscript
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"golang.org/x/crypto/ripemd160"
+)
+
+const (
+	// pubKeyHashSigScriptLen is the length of a signature script attempting
+	// to spend a P2PKH script. This length is determined by the following:
+	//   0x48 (72 byte data push) | <71 byte sig> | <1 byte sighash flag> |
+	//   0x21 (33 byte data push) | <33 byte compressed pubkey>
+	pubKeyHashSigScriptLen = 107
```

Comment:
The signature can either be 71 or 72 bytes, as a result this needs to be a bit fuzzier. Alternatively (in the check below) we can attempt to parse out the final stack items to determine if that's a public key, possibly using pubkey recovery to double check our guess.

Reply:
Fixed. Ended up making the check a bit more lossy and we'll now also attempt to parse the compressed public key found within it. If parsing it fails, we'll fall back to assuming it is a P2SH redeem script.